src/Symfony/Component/Serializer/DependencyInjection/SerializerPass.php

```diff
@@ -50,11 +50,39 @@ public function process(ContainerBuilder $container)
 
         $serializerDefinition = $container->getDefinition($this->serializerService);
         $serializerDefinition->replaceArgument(0, $normalizers);
+        $this->addDefaultContextParameter($normalizers, $container);
 
         if (!$encoders = $this->findAndSortTaggedServices($this->encoderTag, $container)) {
             throw new RuntimeException(sprintf('You must tag at least one service as "%s" to use the "%s" service.', $this->encoderTag, $this->serializerService));
         }
 
         $serializerDefinition->replaceArgument(1, $encoders);
+        $this->addDefaultContextParameter($encoders, $container);
+    }
+
+    private function addDefaultContextParameter($services, $container)
+    {
+        foreach ($services as $service) {
+            $definition = $container->getDefinition($service);
+            if (!$definition->isAutowired()) {
+                continue;
+            }
+
+            if (null === $class = $definition->getClass()) {
+                continue;
+            }
+
+            $reflection = new \ReflectionClass($class);
+
+            if (null === $constructor = $reflection->getConstructor()) {
+                continue;
+            }
+
+            foreach ($constructor->getParameters() as $arg) {
+                if ('defaultContext' === $arg->name) {
+                    $definition->setArgument('$defaultContext', '%serializer.default_context%');
```

Comment:
Maybe only if it's not already set (to allow the user overwrite this)?

Reply:
I may use `bind->('$defaultContext', '...');` in the serializer maybe?

Reply:
Using bindings sounds better, `$definition->setBindings(['$defaultContext', new BoundArgument($defaultContext, false)] + $value->getBindings());`.

Also I'm not fond of exposing configuration values as container parameters, those are not extension points. The parameter should be resolved before injection and be removed from the container once we are done with it.